From 453e3f0bd17a7b1c5afd8f2bfbee796c1225c1a2 Mon Sep 17 00:00:00 2001
From: Hui Tang <tanghui20@huawei.com>
Date: Mon, 30 May 2022 19:55:03 +0800
Subject: [PATCH 2190/2592] [Huawei] sched: Adjust the sysfs path of
 'preferred_cpuset'

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 186873

-------------------------------

Move sysfs of 'preferred_cpuset' from '/proc/pid/' to
'/proc/pid/task/{tid}/'.

Signed-off-by: Hui Tang <tanghui20@huawei.com>
---
 fs/proc/base.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/fs/proc/base.c b/fs/proc/base.c
index 4414b4036e10..8b6d8d04e07a 100644
--- a/fs/proc/base.c
+++ b/fs/proc/base.c
@@ -3416,9 +3416,6 @@ static const struct pid_entry tgid_base_stuff[] = {
 #ifdef CONFIG_CGROUPS
 	ONE("cgroup",  S_IRUGO, proc_cgroup_show),
 #endif
-#ifdef CONFIG_QOS_SCHED_DYNAMIC_AFFINITY
-	REG("preferred_cpuset", 0644, proc_preferred_cpuset_operations),
-#endif
 #ifdef CONFIG_PROC_CPU_RESCTRL
 	ONE("cpu_resctrl_groups", S_IRUGO, proc_resctrl_show),
 #endif
@@ -3807,6 +3804,9 @@ static const struct pid_entry tid_base_stuff[] = {
 #ifdef CONFIG_ASCEND_SHARE_POOL
 	ONE("sp_group", 0444, proc_sp_group_state),
 #endif
+#ifdef CONFIG_QOS_SCHED_DYNAMIC_AFFINITY
+	REG("preferred_cpuset", 0644, proc_preferred_cpuset_operations),
+#endif
 };
 
 static int proc_tid_base_readdir(struct file *file, struct dir_context *ctx)
-- 
2.17.1

